cmake/kokkos_arch.cmake

```diff
@@ -375,6 +376,23 @@ IF (KOKKOS_ARCH_HSW)
   )
 ENDIF()
 
+IF (KOKKOS_ARCH_RISCV_MILKV)
+  IF(NOT (
+  (KOKKOS_CXX_COMPILER_ID STREQUAL GNU
+    AND KOKKOS_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12)
+  OR
+  (KOKKOS_CXX_COMPILER_ID STREQUAL Clang 
+    AND KOKKOS_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 16)
```

Comment:
I made up the compiler versions in the snippet I wrote.  Please check that it works or drop them.

Reply:
Sure, I can only check the gcc one, since I do not have clang installed on my RISC-V machine.

Reply:
Looks like `clang` accepts the flag from version 14 on, see https://godbolt.org/z/M7TGG5eWT.

Reply:
```suggestion
    AND KOKKOS_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 14)
```